### PR TITLE
MM-52834 : Migrate "components/admin_console/dropdown_setting.jsx" and tests to Typescript

### LINKED ISSUE #23426: Migrate "components/admin_console/dropdown_setting.jsx" and tests to Typescript
We are commencing the migration of the mattermost-server/webapp to TypeScript in order to enhance code quality. This "Help Wanted" issue pertains to the migration of the aforementioned "JSX" files and their associated test files to TypeScript.

Below is a checklist outlining the tasks to be performed during the migration:

< ] Rename the files to their corresponding TypeScript extensions (e.g., js to ts, jsx to tsx).
[ ] Update any components that utilize these files with the correct imports.
[ ] Migrate the associated tests, if applicable.
[ > Address any errors raised by the TypeScript compiler. You can execute make check-types to view any encountered errors.

Our objective is to maintain a high level of stringency during this process. However, if you encounter any obstacles where successful migration would necessitate significant changes to other libraries or our tsconfig.json file, please raise the issues. We aim to ensure a smooth and gradual migration while enforcing as many changes as possible.

For reference, you can find examples of already migrated files in the following pull requests:
[https://github.com/mattermost/mattermost-webapp/pull/3790](https://github.com/mattermost/mattermost-webapp/pull/3790|smart-link) 
[https://github.com/mattermost/mattermost-webapp/pull/3472](https://github.com/mattermost/mattermost-webapp/pull/3472|smart-link) 
[https://github.com/mattermost/mattermost-webapp/pull/4230](https://github.com/mattermost/mattermost-webapp/pull/4230|smart-link) 

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-52834

### PR DESCRIPTION
**Pull request by ujwalkumar1995**
#### Summary
This PR is for migration of dropdown_setting.jsx to typescript

#### Ticket Link
Fixes #23426 
JIRA: https://mattermost.atlassian.net/browse/MM-52834

#### Screenshots
NA
#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by ujwalkumar1995** [comment]
Hey Folks
Could someone please review this PR and let me know in case of any feedback.

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx [hunk 1]
```diff
@@ -2,11 +2,22 @@
...
+    values: Array<{text: string; value: string}>;
+    label: ReactNode;
+    value: string;
+    onChange: (id: string, value: string) => void;
+    disabled?: boolean;
+    setByEnv: boolean;
+    helpText: ReactNode;
+}
+
+export default class DropdownSetting extends Component<Props> {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/dropdown_setting.tsx:20]: class DropdownSetting extends React.PureComponent

It was pure components

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx (hunk 1 above)
Comment (ujwalkumar1995) [reply webapp/channels/src/components/admin_console/dropdown_setting.tsx]: Thank you for the review. Updated.

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @ujwalkumar1995!

**Comment by M-ZubairAhmed** [comment]
@ujwalkumar1995 oh darn, can you please fix the conflicts

**Comment by M-ZubairAhmed** [comment]
@ujwalkumar1995 you will need to update your fork to `mattermost` for the CI to pass

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx [hunk 2]
```diff
@@ -59,4 +71,4 @@ export default class DropdownSetting extends React.PureComponent {
             </Setting>
         );
     }
-}
+}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/dropdown_setting.tsx:74]: nit : extra line after this

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by ujwalkumar1995** [comment]
> @ujwalkumar1995 you will need to update your fork to `mattermost` for the CI to pass

Sorry but I did not quite got this, what do I need to do here?

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx (hunk 2 above)
Comment (ujwalkumar1995) [reply webapp/channels/src/components/admin_console/dropdown_setting.tsx]: Fixed

**Comment by M-ZubairAhmed** [comment]
> > @ujwalkumar1995 you will need to update your fork to `mattermost` for the CI to pass
> 
> Sorry but I did not quite got this, what do I need to do here?

Your fork is 
image

it should be `mattermost`. Repo name was changed recently

**Comment by ujwalkumar1995** [comment]
Hey @M-ZubairAhmed 
Thank you for the help and sorry for my limited knowledge in this regard. I have renamed by fork but is this as simple as renaming my fork or do I have to follow some other process?

**Comment by M-ZubairAhmed** [comment]
> Hey @M-ZubairAhmed Thank you for the help and sorry for my limited knowledge in this regard. I have renamed by fork but is this as simple as renaming my fork or do I have to follow some other process?

That should do that, I will monitor the CI if it allows to merge

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**